BDArmory/Armor/ModuleReactiveArmor.cs

```diff
@@ -89,11 +89,14 @@ public void UpdateSectionScales()
                 }
                 if (HP.Hitpoints < 0) part.Destroy();
             }
-            for (int i = 0; i < sectionsCount; i++)
+            if (sectionDestroyed < 0)
+                sectionDestroyed = (int)Mathf.Floor(UnityEngine.Random.value * sectionsRemaining);
```

Comment:
What happens if this picks a section that has already been destroyed?
This needs to keeps a `HashSet<int>` of currently non-destroyed sections. Something like
```csharp
HashSet<int> sectionsIndex = Enumerable.Range(0, numberOfSections).ToHashSet();
...
if (sectionDestroyed < 0)
    sectionDestroyed = sectionsIndex.ElementAt((int)Mathf.Floor(Random.value * sectionsIndex.Count));
sectionsIndex.Remove(sectionDestroyed);
```
`UpdateSectionScales` should probably return `true` if it successfully destroyed a section and `false` otherwise too so that bad things can happen if a destroyed section was hit.

Reply:
Technically it shouldn't be possible since the scale of that piece is being dropped to 0, so the raycast shouldn't return a destroyed section.

Reply:
For the places where it's called from the result of a raycast, yes, but if it's called with the default argument (e.g., from `ExplosionFX`), then the `sectionDestroyed` from the random number is going to repeatedly destroy low indexed sections and hardly ever the highest indexed section.